src/averageAges.js

```diff
@@ -15,11 +15,11 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm'
+    && (!century || Math.ceil(person.died / 100) === century));
+  const totalAge = men.reduce((sum, man) => sum + man.died - man.born, 0);
+
+  return totalAge / men.length;
```

Comment:
move to the separate function & reuse it 
<img width="428" alt="image" src="https://user-images.githubusercontent.com/91411692/227924046-586f54a7-6c9c-44de-af98-05028784b3c2.png">